### PR TITLE
python310Packages.torchaudio: init at 2.0.1

### PR DESCRIPTION
**Pull request by veprbl**
###### Description of changes

We already have `torchaudio-bin`, this adds a source build version (a minimal one, for now).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/torchaudio/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  ];
+
+  BUILD_SOX=0;
+  BUILD_KALDI=0;
+  BUILD_RNNT=0;
+  BUILD_CTC_DECODER=0;
+
+  dontUseCmakeConfigure = true;
+
+  doCheck = false; # requires sox backend
```
Comment (junjihashimoto) [inline pkgs/development/python-modules/torchaudio/default.nix:47]: I think it is better to set these options as arguments of this file.
torchaudio can support CUDA. 
Could you add the option of CUDA?

Comment (veprbl) [reply pkgs/development/python-modules/torchaudio/default.nix:47]: Currently, none of these can be enabled, as they cause build issues that need to be resolved.

The CUDA support is needed, I agree.

**Comment by junjihashimoto** [comment]
I'm worried that without CUDA, users will be late noticing performance issues later.

File: pkgs/development/python-modules/torchaudio/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  meta = with lib; {
+    description = "PyTorch audio library";
+    homepage = "https://pytorch.org/";
+    changelog = "https://github.com/pytorch/audio/releases/tag/v${version}";
+    license = licenses.bsd2;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ junjihashimoto ];
```
Comment (junjihashimoto) [inline pkgs/development/python-modules/torchaudio/default.nix:61]: Oh, I just realized that I am a maintainer.

**Comment by veprbl** [comment]
CUDA should be is working now.

File: pkgs/development/python-modules/torchaudio/default.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/development/python-modules/torchaudio/default.nix:61]: Was a copy-paste from torchaudio-bin. We can change of course.

**Comment by junjihashimoto** [comment]
If so, it is great! Thx!

**Pull request merged**